static/app/views/explore/hooks/useSortByFields.spec.tsx

```diff
@@ -0,0 +1,100 @@
+import {LocationFixture} from 'sentry-fixture/locationFixture';
+import {OrganizationFixture} from 'sentry-fixture/organization';
+
+import {makeTestQueryClient} from 'sentry-test/queryClient';
+import {renderHook} from 'sentry-test/reactTestingLibrary';
+
+import type {Organization} from 'sentry/types/organization';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+import {QueryClientProvider} from 'sentry/utils/queryClient';
+import {useLocation} from 'sentry/utils/useLocation';
+import {PageParamsProvider} from 'sentry/views/explore/contexts/pageParamsContext';
+import {Mode} from 'sentry/views/explore/contexts/pageParamsContext/mode';
+import {SpanTagsProvider} from 'sentry/views/explore/contexts/spanTagsContext';
+import {useSortByFields} from 'sentry/views/explore/hooks/useSortByFields';
+import {OrganizationContext} from 'sentry/views/organizationContext';
+
+jest.mock('sentry/utils/useLocation');
+const mockedUsedLocation = jest.mocked(useLocation);
+
+function createWrapper(organization: Organization) {
+  return function ({children}: {children?: React.ReactNode}) {
+    return (
+      <QueryClientProvider client={makeTestQueryClient()}>
+        <OrganizationContext.Provider value={organization}>
+          <PageParamsProvider>
+            <SpanTagsProvider dataset={DiscoverDatasets.SPANS_EAP} enabled>
+              {children}
+            </SpanTagsProvider>
+          </PageParamsProvider>
+        </OrganizationContext.Provider>
+      </QueryClientProvider>
+    );
+  };
+}
+
+describe('useSortByFields', () => {
+  const organization = OrganizationFixture();
+
+  beforeEach(function () {
+    MockApiClient.addMockResponse({
+      url: `/organizations/org-slug/spans/fields/`,
+      body: [],
+    });
+
+    mockedUsedLocation.mockReturnValue(
+      LocationFixture({query: {dataset: DiscoverDatasets.SPANS_EAP}})
+    );
```

Comment:
Why is this necessary exactly? I'm pretty sure this it defaults to `SPANS_EAP` by default

Reply:
I ran into an issue with the tests - so I mocked LocationFixture with the exact values I needed - but the params are not necessary, so I removed it 👍